### ISSUE TITLE
bug: unable to delete workspace using mixed parameters after upgrade to v17.0

### ISSUE THREAD

**Issue by pratheekrebala** [coder/coder#6088]
Good Afternoon!

I just upgrade to v17.0 and was trying to delete a workspace that also used legacy parameters so I could make the switch to entirely using rich parameters. However, I am unable to delete them using either the CLI or the UI. I receive the following message when trying to remove the workspace:

**CLI**: `coder delete user/workspace`

![image](https://user-images.githubusercontent.com/1174162/217350753-9d4e8cde-db5c-4f34-b63d-35163fa245b1.png)

**UI**: 

![image](https://user-images.githubusercontent.com/1174162/217350926-9e9ed0c4-35b5-4c27-b29b-55b22c535751.png)

I believe this was because I added rich parameters to an existing workspace back in v16.0. Is there any path for me to delete this workspace? (Outside of just deleting with `--orphan` and removing resources manually?)

Thank you,
Pratheek

Related: https://github.com/coder/coder/pull/5974, https://github.com/coder/coder/issues/5574.

**Comment by kylecarbs** [comment]
@pratheekrebala sorry we broke this for you! This was an arbitrary gate on our side but intentional because we plan on forking the UX while users transfer over.

Is there a missing feature in rich parameters that prevents you from moving legacy ones over?

**Comment by pratheekrebala** [comment]
Thanks for the context @kylecarbs!

I don't have any missing features with rich parameters. I do however use `variables` with `sensitive = true` to manage template level parameters.

I ran into this issue while I was working on migrating workspaces to rich parameters. I also tried to delete the workspace using `--orphan` but I get the same error (`Rich paramters can't be used together with legacy parameters.`)

Is there anyway for me to delete the workspace while the gate is in place? I am also unable to update the template associated with the workspace because of the same error. The message on the CLI is different but on the server side I get the same message as before.

Output from `coder template push [template_name] [template_dir]`

**CLI**:

![image](https://user-images.githubusercontent.com/1174162/217382251-a97a2773-041d-4ddd-8702-bb3b26e9fa6e.png)

**UI**:
![image](https://user-images.githubusercontent.com/1174162/217382512-f8666547-1b02-4bea-aff2-addbeb390669.png)
